source/mir/random/package.d

```diff
@@ -307,6 +321,46 @@ version(mir_random_test) unittest
     assert(-double.min_normal < x && x < double.min_normal);
 }
 
+version (LDC)
+{
+    //TODO: figure out specific feature flag or CPU versions where 128 bit multiplication works!
+    version (X86_64)
+        private enum bool probablyCanMultiply128 = true;
+    else
+        private enum bool probablyCanMultiply128 = size_t.sizeof >= ulong.sizeof;
+
+    static if (probablyCanMultiply128 && is(core.simd.Vector!(ulong[2])) && !is(ucent))
+    {
+        private @nogc nothrow pure @safe
+        {
+            pragma(LDC_inline_ir) R inlineIR(string s, R, P...)(P);
+
+            pragma(inline, true)
+            core.simd.ulong2 mul_128(ulong a, ulong b)
+            {
+                return inlineIR!(`
+                    %a = zext i64 %0 to i128
+                    %b = zext i64 %1 to i128
+                    %ra = mul i128 %a, %b
+                    %rb = bitcast i128 %ra to <2 x i64>
+                    ret <2 x i64> %rb`, core.simd.ulong2)(a, b);
```

Comment:
Please do not use SIMD here, they makes code slightly more complicated.
`ulong[2]` can be returned. `(a, b, &c, &d)` where ulong c, d; and return `[a, b]`. It is OK to use 64 bit shift, LLVM is clever enough.

Reply:
In addition you would not need to care Endians